lib/OrganizationsFieldArray/EditOrganizationCard.js

```diff
@@ -0,0 +1,229 @@
+    const { index, organization : { org, id } } = this.props;
+    const isOrgPresent = org && !isEmpty(org);
+    const headerStart = this.renderHeaderStart(isOrgPresent);
+    const cardStyle = isOrgPresent ? 'positive' : 'negative';
+    const name = `${this.props.name}[${index}].org`;
+    const headerEnd = this.renderHeaderEnd(isOrgPresent, id);
+    const key = index;
+    const fieldId = `${this.props.name}-nameOrg-${index}`;
+    const validate = isOrgPresent ? [required, this.validateDuplicateEntries] : required;
+    const renderDefault = !isOrgPresent && this.renderDefaultLayout;
+
+    return {
+      headerStart,
+      cardStyle,
+      index,
+      name,
+      headerEnd,
+      key,
+      id:fieldId,
+      org,
+      validate,
+      renderDefault
+    };
+  };
+
+  renderOrganizationCard = () => (
+    <Field
+      data-test-org-name
+      component={OrganizationCard}
+      {...this.getOrganizationCardProps()}
+    />
+  );
+
+  validateDuplicateEntries = /* istanbul ignore next */ (value, allValues) => {
+    const orgs = allValues[this.props.name] || [];
+
+    const organizations = orgs.filter(organization => {
+      const { org } = organization;
+      return org && (org.name === value.name);
+    });
+
+    if (organizations.length < 2) return undefined;
+    const rolesForThisOrg = organizations.map(org => org.role);
+
+    if (uniq(rolesForThisOrg).length !== rolesForThisOrg.length) {
+      return <FormattedMessage id="stripes-erm-components.organizations.noDuplicates" />;
+    }
+
+    return undefined;
+  }
+
+  renderRole = (name, index) => {
+    return (
+      <Col xs={5}>
+        <Label required tagName="span">
+          <FormattedMessage id="stripes-erm-components.organizations.role" />
+        </Label>
+        <FormattedMessage id="stripes-erm-components.organizations.selectRole">
+          {placeholder => (
+            <Field
+              component={Select}
+              data-test-org-role
+              id={`${name}-role-${index}`}
+              name={`${name}[${index}].role`}
+              placeholder={placeholder}
+              dataOptions={this.props.roles}
+              validate={required}
+            />
+          )}
+        </FormattedMessage>
+      </Col>
+    );
+  }
+
+  renderOrganizationPicker = (index, name) => {
+    return (
+      <this.props.organizationPickerComponent
+        data-test-org-name
+        onAddOrg={this.onAddOrg}
+        index={index}
+        id={`${name}-nameOrg-${index}`}
+      />
+    );
+  }
+
+  renderDefaultLayout = () => (
+    <div>
+      <Layout className="textCentered">
+        <strong>
+          <FormattedMessage id="stripes-erm-components.organizations.noOrganization" />
+        </strong>
+      </Layout>
+      <Layout className="textCentered">
+        <FormattedMessage id="stripes-erm-components.organizations.addOrganizationToStart" />
+      </Layout>
+    </div>
+  );
+
+  onAddOrg = (organization, index) => {
```

Comment:
We've been trying to keep a pattern of callback handlers being called `handleFoo` and being passed as `onFoo`. So this would be called `handleAddOrg` and the below would be called `handleUnlinkOrg`. It makes it easier to track where something is defined rather than where it's just being used or passed through.